edkrepo/common/common_repo_functions.py

```diff
@@ -908,17 +908,19 @@ def _list_patchset_ancestors(manifest, patchset):
 
 def _checkout_parent(patchset, name, repo, parent_patchsets):

```

Comment:
I may have forgotten that "if parent_patchsets" is only run for daisy-chained PatchSet.  The exception should be moved so that it runs also when a non-daisy-chained PatchSet uses a branch like 'master' as the parentSha in the manifest.
  
    def _checkout_parent(patchset, name, repo, parent_patchsets):
        if parent_patchsets:
            base_parent_sha = parent_patchsets[-1].parent_sha
        else:
            base_parent_sha = patchset.parent_sha
    
        if base_parent_sha in repo.tags:
            # tag names are prioritized over branch names
            repo.git.checkout("refs/tags/{}".format(base_parent_sha), b=name)
        elif base_parent_sha in repo.heads:
            # PatchSet should not use a branch name as the base parent_sha.
            raise EdkrepoBranchCollidesWithParentShaException(BRANCH_COLLIDES_WITH_PARENT_SHA)
        else:
            repo.git.checkout(base_parent_sha, b=name)